File: 251004/Yelkin/4-5/TopicService/src/main/java/by/yelkin/TopicService/mapper/MarkMapper.java
package by.yelkin.TopicService.mapper;

import by.yelkin.TopicService.dto.mark.MarkRequestTo;
import by.yelkin.TopicService.dto.mark.MarkResponseTo;
import by.yelkin.TopicService.entity.Mark;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface MarkMapper {
    Mark toEntity(MarkRequestTo markRequestTo);
    MarkResponseTo toResponse(Mark tag);
}

File: 251003/Kukhatskavolets/microblog/src/main/kotlin/by/kukhatskavolets/mappers/MarkMapper.kt
package by.kukhatskavolets.mappers

import by.kukhatskavolets.dto.requests.MarkRequestTo
import by.kukhatskavolets.dto.responses.MarkResponseTo
import by.kukhatskavolets.entities.Mark

fun MarkRequestTo.toEntity() = Mark(
    id = this.id,
    name = this.name
)

fun Mark.toResponse() = MarkResponseTo(
    id = this.id,
    name = this.name
)